src/tools/r2rdump/SignatureType.cs

```diff
@@ -38,20 +43,26 @@ public SignatureType(ref BlobReader signatureReader, MetadataReader mdReader, Ge
         {
             SignatureTypeCode signatureTypeCode = signatureReader.ReadSignatureTypeCode();
             Flags = 0;
-            if (signatureTypeCode == SignatureTypeCode.SZArray)
+
+            if (signatureTypeCode == SignatureTypeCode.ByReference)
```

Comment:
This could be a good area for a switch statement, with the default case doing nothing.

Reply:
If the param is a ref to an array, it would go into the ByReference condition and read the next typecode which is array, then go into the SZArray condition. With a switch statement, it wouldn't go into both conditions

Reply:
Oh I see, I didn't recognize that you read the SignatureTypeCode again in this block. Disregard the comment then.